### PR TITLE
web_service: Port yuzu-emu/yuzu#1465 and Port yuzu-emu/yuzu#1473 (Many self-contained changes)

### PR DESCRIPTION
**Pull request by FearlessTobi**
See yuzu-emu/yuzu#1465 and yuzu-emu/yuzu#1473 for more details.

It's highly advised to review commit-by-commit.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/4359)

### CODE REVIEW

File: src/web_service/telemetry_json.cpp [hunk 1]
```diff
@@ -3,8 +3,8 @@
 // Refer to the license.txt file included.
 
 #include <json.hpp>
+#include "common/announce_multiplayer_room.h"
```
Comment (wwylele) [inline src/web_service/telemetry_json.cpp]: :confused: What does announce_multiplayer_room have to do with telemetry? Please #include the real header you need.

Comment (wwylele) [reply src/web_service/telemetry_json.cpp]: @FearlessTobi Then do it like yuzu: move it to its own file.

**Comment by FearlessTobi** [comment]
Build failure is not related to the PR.

File: src/web_service/web_backend.cpp [hunk 2]
```diff
@@ -10,6 +10,7 @@
 #include "common/announce_multiplayer_room.h"
```
Comment (wwylele) [inline src/web_service/web_backend.cpp]: You can now remove this.

**Pull request merged**